raw/fetch.py

```diff
@@ -0,0 +1,224 @@
+API_KEY_PATTERN = re.compile(r'^API_KEY_\d+$')
+API_KEY_LIST = []
+
+
+def new_config_file(config_file: str, field_name: str, additional_content: str,
+    co_file: bool, single_level_file: bool, first_day_first_prev: datetime.date,
+    last_day_first_prev: datetime.date, first_day_third_prev: datetime.date,
+    last_day_third_prev: datetime.date, sl_year: str, sl_month: str,
+    sl_first_date: str, sl_last_date: str) -> None:
+    """Modify the specified configuration file with new values.
+
+    Parameters:
+        config_file (str): The path to the configuration file to be modified.
+        field_name (str): The name of the field to be updated with the new value.
+        additional_content (str): The additional content to be added under the 
+                                    '[selection]' section.
+        co_file (bool): True if the configuration file is a 'hourly' file, False 
+                        otherwise.
+        single_level_file (bool): True if the configuration file is a 'sfc' file,
+                                 False otherwise.
+        first_day_first_prev (datetime.date): The first day of the first previous month.
+        last_day_first_prev (datetime.date): The last day of the first previous month.
+        first_day_third_prev (datetime.date): The first day of the third previous month.
+        last_day_third_prev (datetime.date): The last day of the third previous month.
+        sl_year (str): The year of the third previous month in 'YYYY' format.
+        sl_month (str): The month of the third previous month in 'MM' format.
+        sl_first_date (str): The first date of the third previous month in 'DD' format.
+        sl_last_date (str): The last date of the third previous month in 'DD' format.
+    """
+
+    with open(config_file, "r") as file:
+        lines = file.readlines()
+
+    # Update the specified field with the new value
+    updated_lines = []
+    selection_line_found = False
+    for line in lines:
+        if not selection_line_found and line.strip() == "[selection]":
+            updated_lines.append(f"{additional_content}\n")
+            selection_line_found = True
+
+        if single_level_file:
+            if line.startswith("year"):
+                line = f"year={sl_year}\n"
+            if line.startswith("month"):
+                line = f"month={sl_month}\n"
+            if line.startswith("day"):
+                line = f"day={sl_first_date}/to/{sl_last_date}\n"
+        elif line.startswith(field_name):
+            if co_file:
+                line = (
+                    f"{field_name}={first_day_first_prev}/to/{last_day_first_prev}\n"
+                )
+            else:
+                line = (
+                    f"{field_name}={first_day_third_prev}/to/{last_day_third_prev}\n"
+                )
+        updated_lines.append(line)
+
+    with open(config_file, "w") as file:
+        file.writelines(updated_lines)
```

Comment:
Instead of write lines (which seems brittle to me), consider reading and writing the files with the `configparser`: https://docs.python.org/3/library/configparser.html

I think it would be easier to manipulate than manipulating lines of strings, too.

Reply:
Code updated as per your suggestion. Thanks for the input!